### DISCUSSION TITLE
OpenAPI specification field name capitalisation doesn't match

### DISCUSSION THREAD

**Discussion by CharlieArmstrongDev** [ServiceStack/Discuss#26] [Q&A] [answered]
Hey all, anyone got experience with the openAPI specification feature?
I've been trying to hook all the CRUD endpoints into RTK Query, which has it's own code auto-generation that can hook into OpenAPI.

Some awkward result of something in ServiceStack means that while everything in the OpenAPI spec matches the original database fields capitalisation, just like the C# definition of the DTOs. So when I run the code generation, all the types in my typescript frontend match this too.
However, what is actually being sent over the server to the react hooks is an object where the capitalisation has been altered, to be the same as how it'd be in the TypeScript generated DTOs.
![image](https://user-images.githubusercontent.com/97681489/181650791-84d72744-56f1-4055-a1eb-71d0b3ba15a3.png)
![image](https://user-images.githubusercontent.com/97681489/181650920-4f641141-732c-4a60-89be-f6daffc1220e.png)

Even in the Swagger UI (which comes with the OpenAPI feature) you can see the difference in what it expects and what it gets. Does anyone with some more experience have more of an idea what is happening here? I suppose it's something to do with JSON maybe?
![image](https://user-images.githubusercontent.com/97681489/181651076-9295e374-e5d2-4093-9fb5-a59174cf2a69.png)

**Comment by mythz** [comment]
I'd recommend using the [TypeScript Add ServiceStack Reference](https://docs.servicestack.net/typescript-add-servicestack-reference) instead which is a feature we have total control over that results in much cleaner code-gen + dev model.

**Comment by CharlieArmstrongDev** [comment, marked as answer]
Coming back to this just in-case anyone else ever has the same problem - I had missed the option to set this to true: 
![image](https://user-images.githubusercontent.com/97681489/186630188-696c2fe3-870f-4f8a-9961-a9c167748801.png)
